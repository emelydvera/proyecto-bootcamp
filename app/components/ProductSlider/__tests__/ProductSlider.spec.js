const React = require('react');
const ProductSlider = require('..');
const { render, screen, fireEvent } = require('@testing-library/react');
const { I18n, I18nProvider } = require('nordic/i18n');
const { getByAltText } = screen;


const { data } = require('../../../../mocks/test/get/https/api.mercadolibre.com/items/MLA1104983845.json')

const i18n = new I18n({
    translations: {}
})

describe('<ProductSlider/>', () => {

    let component;
    const product = {
        title: data.title,
        pictures: data.pictures
    }

    beforeEach(() => {

        component = render(
            <I18nProvider i18n={i18n}>
                <ProductSlider product={product} />
            </I18nProvider>
        )

    })

    it('should render the components', () => {

        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();

    });

    it('should change the parent image to the one that fires the event', () => {

        const img = document.querySelector("img[data-src='https://http2.mlstatic.com/D_887836-MLA47152640900_082021-O.jpg']");
        const mainImage = getByAltText('product');

        expect(mainImage).toHaveAttribute('src', 'https://http2.mlstatic.com/D_852458-MLA48270995220_112021-O.jpg')
        
        fireEvent.mouseOver(img);

        expect(mainImage).toHaveAttribute('src', 'https://http2.mlstatic.com/D_887836-MLA47152640900_082021-O.jpg')

    });

})