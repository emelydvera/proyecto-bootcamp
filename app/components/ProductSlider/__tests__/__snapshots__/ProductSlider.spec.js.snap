// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductSlider/> should render the components 1`] = `
<DocumentFragment>
  <div
    class="slider__img"
  >
    <div
      class="slider__img__carousel"
    >
      <div
        class="image__container active"
      >
        <img
          alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
          aria-label="imagen del producto"
          class="slider__image  lazy-loadable"
          data-src="https://http2.mlstatic.com/D_852458-MLA48270995220_112021-O.jpg"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <div
        class="image__container "
      >
        <img
          alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
          aria-label="imagen del producto"
          class="slider__image  lazy-loadable"
          data-src="https://http2.mlstatic.com/D_887836-MLA47152640900_082021-O.jpg"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <div
        class="image__container "
      >
        <img
          alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
          aria-label="imagen del producto"
          class="slider__image  lazy-loadable"
          data-src="https://http2.mlstatic.com/D_992922-MLA48271038089_112021-O.jpg"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <div
        class="image__container "
      >
        <img
          alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
          aria-label="imagen del producto"
          class="slider__image  lazy-loadable"
          data-src="https://http2.mlstatic.com/D_636041-MLA48271038093_112021-O.jpg"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <div
        class="image__container "
      >
        <img
          alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
          aria-label="imagen del producto"
          class="slider__image  lazy-loadable"
          data-src="https://http2.mlstatic.com/D_827704-MLA47152858202_082021-O.jpg"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
      <div
        class="image__container "
      >
        <img
          alt=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
          aria-label="imagen del producto"
          class="slider__image  lazy-loadable"
          data-src="https://http2.mlstatic.com/D_935467-MLA48270794800_112021-O.jpg"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        />
      </div>
    </div>
    <figure
      aria-label="Imagen de  Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
      tabindex="9"
    >
      <img
        alt="product"
        class="slider__img__main"
        decoding="async"
        src="https://http2.mlstatic.com/D_852458-MLA48270995220_112021-O.jpg"
      />
      <figcaption>
         Moto G20 64 Gb  Azul Cielo 4 Gb Ram
      </figcaption>
    </figure>
  </div>
</DocumentFragment>
`;
